tests/Opus/Search/Common/IndexerTest.php

```diff
@@ -0,0 +1,961 @@
+	 */
+	public function testMissingConfigParamSearchEngine_Index_Host() {
+		$this->markTestSkipped( 'not supported in Opus_Search API due to passing configuration to 3rd-party library for processing' );
+
+		$this->adjustConfiguration( array(), function ( Zend_Config $config ) {
+			unset(
+				$config->searchengine->solr->default->service->index->endpoint->primary->host,
+				$config->searchengine->solr->default->service->default->endpoint->primary->host
+			);
+
+			return $config;
+		} );
+
+		Opus_Search_Service::selectIndexingService()
+		                   ->removeAllDocumentsFromIndex();
+	}
+
+	/**
+	 * @expectedException Opus_Search_InvalidConfigurationException
+	 */
+	public function testMissingConfigParamSearchEngine_Index_Port() {
+		$this->markTestSkipped( 'not supported in Opus_Search API due to passing configuration to 3rd-party library for processing' );
+
+		$this->adjustConfiguration( array(), function ( Zend_Config $config ) {
+			unset(
+				$config->searchengine->solr->default->service->index->endpoint->primary->port,
+				$config->searchengine->solr->default->service->default->endpoint->primary->port
+			);
+
+			return $config;
+		} );
+
+		Opus_Search_Service::selectIndexingService()
+		                   ->removeAllDocumentsFromIndex();
+	}
+
+	/**
+	 * @expectedException Opus_Search_InvalidConfigurationException
+	 */
+	public function testMissingConfigParamSearchEngine_Index_Path() {
+		$this->markTestSkipped( 'not supported in Opus_Search API due to passing configuration to 3rd-party library for processing' );
+
+		$this->adjustConfiguration( array(), function ( Zend_Config $config ) {
+			unset(
+				$config->searchengine->solr->default->service->index->endpoint->primary->path,
+				$config->searchengine->solr->default->service->default->endpoint->primary->path
+			);
+
+			return $config;
+		} );
+
+		Opus_Search_Service::selectIndexingService()
+		                   ->removeAllDocumentsFromIndex();
+	}
+
+	public function testMissingConfigParamLogPrepareXml() {
+		$this->adjustConfiguration( array(), function ( Zend_Config $config ) {
+			unset( $config->log->prepare->xml );
+
+			return $config;
+		} );
+
+		$this->_addOneDocumentToIndex();
+	}
+
+	/**
+	 * @expectedException Opus_Search_InvalidConfigurationException
+	 */
+	public function testEmptyConfiguration() {
+		$this->markTestSkipped( 'not supported in Opus_Search API due to passing configuration to 3rd-party library for processing' );
+
+		$this->adjustConfiguration( array(
+            'searchengine' => array( 'solr' => array( 'default' => array( 'service' => array( 'index' => array( 'endpoint' => array( 'primary' => array( 'path' => '' ) ) ) ) ) ) )
+        ) );
+
+		Opus_Search_Service::selectIndexingService()
+		                   ->removeAllDocumentsFromIndex();
+	}
+
+	/**
+	 * @expectedException Opus_Search_InvalidConfigurationException
+	 */
+	public function testInvalidConfiguration() {
```

Comment:
I agree that it is not really a unit test.  It is more like a integration test. We have a lot of those. It does not make sense as a Selenium test. As a "unit test" it takes milliseconds while a Selenium test clocks in at a minimum of 6-9 seconds. So Selenium is only for tests that cannot be done otherwise. I don't really mind were the tests are located, but  I would like something that shows what happens when the configuration is messed up. Some of the old tests show that nothing happens. OPUS 4 (so far) is supposed to run even if the Solr Server is missing. Of course the index will become out of date, but the Application is supposed to still work. I think the error handling has changed with the new API. There are for instance all the new error messages when indexing the test data.